hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruBlockCache.java

```diff
@@ -947,7 +1019,47 @@ public void run() {
         }
         LruBlockCache cache = this.cache.get();
         if (cache == null) break;
-        cache.evict();
+        bytesFreed = cache.evict();
+        long stopTime = System.currentTimeMillis();
+        // If heavy cleaning BlockCache control.
+        // It helps avoid put too many blocks into BlockCache
+        // when evict() works very active.
+        if (stopTime - startTime <= 1000 * 10 - 1) {
```

Comment:
10s because this is into synchronized section above:
  this.wait(1000 * 10/*Don't wait for ever*/);

The eviction can start sometimes more often sometimes rarely. This section helps lead to a similar period. Provided more details into comments.

Reply:
Oh I see.. looks like you intend to compute every 10s, but the problem is evictions can happen out of sync (look for callers of evict). For example in a timeline, it can happen at 10s, 12s, 23s etc. So at the end of 23s the mbs computed correspond to a total 13s (between 1st and last)..